### ISSUE TITLE
(t) Rockstor 4.5.7 will offer upgrade, though there is none

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2504]
Running 4.5.7 on a virtual machine with amd86_64 architecture. The WebUI shows that an upgrade is available:
<img width="311" alt="image" src="https://user-images.githubusercontent.com/35113775/221983227-e8eb18c5-3be1-4b9e-9de0-08d2c1095d23.png">

When selecting the up arrow, it takes you to the `version` page (i.e. stable/testing channel selection). It shows this message below:
<img width="1081" alt="image" src="https://user-images.githubusercontent.com/35113775/221983558-a6adb3c9-198a-4f5e-b487-29cafaae3e8b.png">

When then selecting the `Start Update` button, the update seems to start:
<img width="548" alt="image" src="https://user-images.githubusercontent.com/35113775/221983710-eb3c02e1-c966-4710-b821-88d31af220b1.png">

After a few minutes it indicates the update is still going on, and then after some more time, the screen refreshes back to the channel display, but nothing has changed (as it shouldn't I think).

Nothing happening in the rockstor logs.

_On a side note: while in the `/opt/rockstor/var/log directory` I noticed that there are 2 ztask log files (size zero) that were created during the installation. I assume, because of the change to huey, those won't be needed anymore (and I can see a huey log in the same directory)? Or are those there, because more cleanup of ztask needs to happen before those won't be necessary anymore? If that is reporting worthy issue, I can create a separate one for that._

<img width="476" alt="image" src="https://user-images.githubusercontent.com/35113775/221986013-141b864a-f79d-419a-949e-7a8cab7d649b.png">

**Comment by phillxnet** [comment]
@Hooverdan96 This issue looks very much like the following one on our current Milestone of:
First Stable Poetry build: https://github.com/rockstor/rockstor-core/milestone/19
I.e.
Duplicate of: #2468
(t) alt arch causing false positive update available #2468


> I assume, because of the change to huey, those won't be needed anymore (and I can see a huey log in the same directory)? Or are those there, because more cleanup of ztask needs to happen before those won't be necessary anymore? If that is reporting worthy issue, I can create a separate one for that.

Yes, this should should be another issue I think. But all of ztaskd was removed when we moved to Huey. But to preserve things like database entries, service names etc, I, in many places, simply carried on with the ztaskd naming as it was then a far less disruptive upgade: doing away with a bunch of stuff that would have been required just to call something huey rather than ztaskd. So I did leave a bunch of naming unchanged. Not sure, of the top of my head, where these logs are from however but likely they are similarly a preserve-naming move where the ztask-daemon actually relates to the huey daemon. But as you say it has it's own logging also. A seperate issue could be used to resolve this for sure and potentially remove them to avoid accidental rabit hole and make on-boarding just that little bit more challenging.

Thanks for the report here. But to me it looks very much like the declared duplicate. If you agree then we can close this as a duplicate as the original has a fair bit of into in already as to the cause of this false positive on update. And as you observed and commented, it's mainly a rather unsettling cosmetic of no functional consequence (but significant psychological consequence of course :) hence it being on the current Milestone for our next Stable release.

**Comment by Hooverdan96** [comment]
Well, here I thought I found something cosmetically special, and only scanned the open items to avoid a duplication. I will close this one, and open another one for the ztaskd piece, you can disposition that one then when there's nothing else left to do 👍
